httpx/streams.py

```diff
@@ -0,0 +1,297 @@
+import binascii
+import mimetypes
+import os
+import typing
+from io import BytesIO
+from json import dumps as json_dumps
+from pathlib import Path
+from urllib.parse import urlencode
+
+from .utils import format_form_param
+
+RequestData = typing.Union[dict, str, bytes, typing.AsyncIterator[bytes]]
+
+RequestFiles = typing.Dict[
+    str,
+    typing.Union[
+        # file (or str)
+        typing.Union[typing.IO[typing.AnyStr], typing.AnyStr],
+        # (filename, file (or str))
+        typing.Tuple[
+            typing.Optional[str], typing.Union[typing.IO[typing.AnyStr], typing.AnyStr],
+        ],
+        # (filename, file (or str), content_type)
+        typing.Tuple[
+            typing.Optional[str],
+            typing.Union[typing.IO[typing.AnyStr], typing.AnyStr],
+            typing.Optional[str],
+        ],
+    ],
+]
+
+
+class Stream:
+    async def __aiter__(self) -> typing.AsyncIterator[bytes]:
+        yield b""
+
+    async def aclose(self) -> None:
+        pass
+
+    async def aread(self) -> bytes:
+        return b"".join([part async for part in self])
```

Comment:
We never actually use this method on responses — only on requests.

So should this be moved to `RequestStream`?

This would mean `Stream` ends up being equivalent to `AsyncIterator[bytes]` — which sounds good to me — and resolve a few type hint issues (we don't currently catch) in our tests.

Reply:
Dropped this entirely.